src/server/engine_shard.cc

```diff
@@ -735,11 +735,14 @@ void EngineShard::Heartbeat() {
   DbSlice& db_slice = namespaces->GetDefaultNamespace().GetDbSlice(shard_id());
   // Skip heartbeat if we are serializing a big value
   static auto start = std::chrono::system_clock::now();
-  if (db_slice.WillBlockOnJournalWrite()) {
+  // We acquire exlusive locks during global transactions
```

Comment:
Lets update the comment, 
How about: Skip heartbeat if global transaction is in process. This is determined by attempting to check if shard lock can be acquired.